crates/edr_provider/src/requests/eth/transactions.rs

```diff
@@ -236,7 +237,50 @@ pub fn handle_send_raw_transaction_request(
     data: &mut ProviderData,
     raw_transaction: Bytes,
 ) -> Result<B256, ProviderError> {
-    data.send_raw_transaction(raw_transaction.as_ref())
+    let mut raw_transaction: &[u8] = raw_transaction.as_ref();
+    let signed_transaction =
+        SignedTransaction::decode(&mut raw_transaction).map_err(|err| match err {
+            edr_eth::rlp::Error::Custom(message) if SignedTransaction::is_invalid_transaction_type_error(message) => {
+                ProviderError::InvalidTransactionType(*raw_transaction.first().expect("We already validated that the transaction is not empty if it's an invalid transaction type error."))
```

Comment:
It took me a moment to figure out what was going on here. It might be good to self-document by assigning the first byte to a local variable `let type_id = ...` before passing it to the enum variant

Reply:
Thanks, done in [Make code more eloquent](https://github.com/NomicFoundation/hardhat/pull/4723/commits/1b2b4a9d6c76e672d0fb605e494a53ee9566ff3c)